devito/passes/clusters/implicit.py

```diff
@@ -236,8 +221,8 @@ def _lower_msd(dim, cluster):
 @_lower_msd.register(MultiSubDimension)
 def _(dim, cluster):
     i_dim = dim.implicit_dimension
-    mapper = {(dim.root, i): dim.functions[i_dim, mM]
-              for i, mM in enumerate(dim.bounds_indices)}
+    mapper = {t[0]: dim.functions[i_dim, mM]
```

Comment:
@EdCaunt this change broke PRO's sibling PR, will fix

Reply:
also, we should avoid magic numbers at all costs, and 0 here is a magic number

Reply:
I've got a fix for it already. Will push it